core/deployment/src/main/java/io/quarkus/deployment/pkg/PackageConfig.java

```diff
@@ -27,6 +29,16 @@
     @ConfigItem(defaultValue = JAR)
     public String type;
 
+    /**
+     * Experimental property that allows for the creation of a 'mutable' application that
+     * can be re-augmented after creation.
+     *
+     * TODO: should this just be a different package type: 'mutable-jar'
```

Comment:
+1 to make it a different package type, since IIUC you can't really use `mutableApplication` with the other package types